File: engine/model/src/main/scala/tools/packet/pet/PetCommandResponse.scala
package tools.packet.pet

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PetCommandResponse(private var _characterId: Int, private var _index: Byte, private var _talk: Boolean,
                         private var _animation: Int, private var _balloonType: Boolean) extends PacketInput {
  def characterId: Int = _characterId

  def index: Byte = _index

  def talk: Boolean = _talk

  def animation: Int = _animation

  def balloonType: Boolean = _balloonType

  override def opcode(): SendOpcode = SendOpcode.PET_COMMAND
}

File: engine/engine-model/src/main/java/tools/packet/pet/PetCommandResponse.java
package tools.packet.pet;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PetCommandResponse(int characterId, byte index, boolean talk, int animation,
                                 boolean balloonType) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PET_COMMAND;
   }
}